src/main/java/org/terracotta/statistics/TableValueStatistic.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics;
+
+import org.terracotta.statistics.extended.StatisticType;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * @author Mathieu Carbou
+ */
+public class TableValueStatistic implements ValueStatistic<Table> {
+
+  public static TableValueStatistic.Builder newBuilder() {
```

Comment:
This is weird. I don't know the solution but we seem to have builders of builders with `Table` and `TableValueStatistic`

Reply:
there is 2 way to provide a statistic of type table. Either by taking a snapshot of the data and returning a table, this works if the data is correlated (i.e. total_count, total_time), or we can directly setup some suppliers for each kind of statistic inside the table if the data is not correlated.

Reply:
Moved to PR #41